src/reducers/importText.ts

```diff
@@ -107,13 +107,20 @@ const importText = (
     const textNormalized = strip(convertedText, { preserveFormatting: true })
 
     // insert the textNormalized into the destValue in the correct place
-    // trim after concatenating in case destValue has whitespace
-    const left = (destValue.slice(0, replaceStart ?? 0) + textNormalized).trimLeft()
     // if cursorCleared is true i.e. clearThought is enabled we don't have to use existing thought to be appended
-    const right = state.cursorCleared ? '' : destValue.slice(replaceEnd ?? 0).trimRight()
-    const newValue = left + right
-
-    const offset = getTextContentFromHTML(left).length
+    const replacedDestValue = state.cursorCleared
+      ? ''
+      : destValue.slice(0, replaceStart || 0) + destValue.slice(replaceEnd || 0)
+
+    const caretPosition =
+      (replaceStart || window.getSelection()?.focusNode?.nodeType === Node.TEXT_NODE
```

Comment:
In order to maintain compatibility with other platforms (i.e. React Native), we avoid directly accessing the DOM. We use `selection.ts` to hide selection access and allow other platforms to shim their own API.

You can use the existing `isText` method from the `selection` module:

https://github.com/cybersemics/em/blob/d225ba99ee9eed5a92b31913fc87f116f19ed2cd/src/device/selection.ts#L109-L110

Reply:
Thanks! I missed out on this part